test/test_sparse.py

```diff
@@ -3248,33 +3025,146 @@ def test_cuda_sparse_cpu_dense_add(self):
         with self.assertRaisesRegex(RuntimeError, "add: expected 'self' to be a CUDA tensor, but got a CPU tensor"):
             x + sparse_y
 
+
 class TestSparseUnaryUfuncs(TestCase):
     exact_dtype = True
 
-    @ops(sparse_unary_ufuncs)
-    def test_sparse_consistency(self, device, dtype, op):
-        unsupportedTypes = [torch.bfloat16, torch.cfloat, torch.cdouble]
+    def _skip_helper(self, dtype):
+        unsupportedTypes = [torch.bfloat16, torch.float16, torch.bool, torch.complex64, torch.complex128]
         if dtype in unsupportedTypes:
-            self.skipTest('Skipped! Unsupported dtypes for Sparse')
+            self.skipTest(f'Skipped! "coalesce" not implemented for {dtype}')
+
+    def _get_dense_and_sparse_samples(self, device, dtype, op):
+        test_backward = op.test_complex_grad or not dtype.is_complex
+        samples = op.sample_inputs(device, dtype, requires_grad=test_backward)
+        sparse_samples = op.sparse_sample_inputs(device, dtype, requires_grad=test_backward)
+        if sparse_samples is not None:
+            samples += sparse_samples
+        return samples
+
+    def _test_forward(self, sample, op):
+        dense_sample_input = tuple(
```

Comment:
This assumes that args and kwargs don't need to be modified between dense and sparse inputs, right? We should make these assumptions explicit (in a comment).

Reply:
Yes, it is rather strong assumption I was making as that the test case is for unary ops only (class name TestSparseUnaryUfuncs) and for which args/kwargs wont be tensors.
Anyway, I'll put that into a comment.